[18.0][MIG] report_wkhtmltopdf_param: Migration to 18.0.

LucasTran380381 opened the pull request:
**Change in v18:**

- replace security empty group to `base.group_user`
- improve test-case: `test_wkhtmltopdf_incorrect_parameter` to catch warning log (OCA 18.0 not allow warning log)
- Due to the deprecation of the `_check_recursion` function in Odoo 18, the method has been renamed to `_check_recursion_custom_params()`.
- refactor bodies in method `_run_wkhtmltopdf` should be a list

Commit:
Rename base_condition_template to base_comment_template

Commit:
[MIG] base_comment_template to 9.0

legalsylvain committed:
new module bi_sql_editor [IMP] function to guess model for many2one field; [ADD] security

Commit:
[ADD] module 'report_wkhtmltopdf_param'

oca-transbot committed:
OCA Transbot updated translations from Transifex

clb-openfire committed:
[MIG] base_comment_template, invoice_comment_template: Migration to 10.0

Port modules base_comment_template and invoice_comment_template to V10

nicomacr committed:
bi_sql_editor

legalsylvain committed:
create ui now create tree view [IMP] fr translation [IMP] add last refresh date time on the action name of each sql materialized view

AaronHForgeFlow committed:
[IMP] Add test + several fixes

nicomacr committed:
FIX bi_sql_editor

oca-transbot committed:
OCA Transbot updated translations from Transifex

serpentcs-dev1 committed:
[MIG] bi_sql_editor v10

serpentcs-dev1 committed:
[IMP] preview button

etobella committed:
[MIG] report_wkhtmltopdf_param: Migraton to 11.0

Commit:
Fix several issues

Fix error '... is not a table or foreign table'
Fix view (colors + oe_highlight)
Fix tests
Fix README + manifest
Fix back to draft
Fix cron call + default values
Use Postgres version 9.6 for travis builds

legalsylvain committed:
10.0 imp bi sql editor (#1)

* [IMP] is_materialized field non readonly on sql_valid state ; [FIX] block possibility to set indexes on non materialized view

* [FIX] set domain_force, group_ids readonly if state > sql_valid

* [IMP] better display of the field group_ids

* [IMP] possibility to reorder menu items from sql views

* [IMP] Do not warn user when setting sql view to draft if state is sql_valid

* [REF]

* [FIX] Set Date of the first execution in the action name

oca-travis committed:
[UPD] Update bi_sql_editor.pot

HviorForgeFlow committed:
[MIG] bi_sql_editor: Migration to 11.0

xavierjimenez committed:
[MIG] base_comment_template: Migration to 11.0

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

HviorForgeFlow committed:
[FIX] Can be upgraded with materialized views generated

MiquelRForgeFlow committed:
[FIX] ir.cron relation

MiquelRForgeFlow committed:
[IMP] remove unneeded parenthesis

MiquelRForgeFlow committed:
[FIX] Solves the upgrading problem

MiquelRForgeFlow committed:
[FIX] Query and technical name  should only be modifiable in draft mode

oca-travis committed:
[UPD] Update bi_sql_editor.pot

[UPD] Update bi_sql_editor.pot

[UPD] Update bi_sql_editor.pot

[UPD] Update bi_sql_editor.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

AdriaGForgeFlow committed:
[12.0][MIG] Migrate module bi_sql_editor to v12

rven committed:
[MIG] base_comment_template: Migration to 12.0

MiquelRForgeFlow committed:
[FIX] bi_sql_editor: add in readme how to solve the update all break

AdriaGForgeFlow committed:
Update readme

mvrodriguez committed:
Translated using Weblate (Galician)

Currently translated at 100.0% (15 of 15 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/gl/

[ADD] icon.png

OCA-git-bot committed:
bi_sql_editor 12.0.1.2.0

bi_sql_editor 12.0.1.1.0

[UPD] README.rst

[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-12.0/reporting-engine-12.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-bi_sql_editor/

Commit:
[12.0][MIG] report_wkhtmltopdf_param

Commit:
[IMP] base_comment_template multi company rules

[IMP] Move comment_template_id from account_invoice_comment_template and make it company_dependant

[IMP] base_comment_template: Add test

Commit:
[IMP] account_invoice_comment_template: Move comment_template_id field to the Invoicing tab

[IMP] account_invoice_comment_template: rename partner field name from comment_template_id to invoice_comment_template_id

[IMP] account_invoice_comment_template: Make partner field company_dependant and move domain definition of invoice fields from the view to the model

[MOV] account_invoice_comment_template: comment_template_id to base_comment_template

[IMP] account_invoice_comment_template: Translate templates when partner changes

Commit:
[MOV] migration script to base_comment_template

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/

rschnapka committed:
Translated using Weblate (German)

Currently translated at 29.1% (32 of 110 strings)

Translation: reporting-engine-12.0/reporting-engine-12.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-bi_sql_editor/de/

yostashiro committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (15 of 15 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/ja/

pedrobaeza committed:
[FIX] bi_sql_editor: Fixes images path

Fixes #309

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (15 of 15 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/pt/

HviorForgeFlow committed:
[FIX] bi_sql_editor: Add required attribute name on filters

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (15 of 15 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/pt_BR/

HviorForgeFlow committed:
change SQL mapping for date

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (21 of 21 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/pt_BR/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (21 of 21 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/pt/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 95.2% (20 of 21 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/hr/

pedrobaeza committed:
[FIX] base_comment_template: Make field truly company dependent

Previous field declaration was incorrect due to a typo, leading to non applying the
company dependent features.

Now everything is correct and the migration scripts care of moving data properly.

pedrobaeza committed:
[FIX] base_comment_template: Make field truly company dependent

Previous field declaration was incorrect due to a typo, leading to non applying the
company dependent features.

Now everything is correct and the migration scripts care of moving data properly.

legalsylvain committed:
[ADD] action context

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (21 of 21 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/de/

pedrocs-exo committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (21 of 21 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/pt_PT/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-12.0/reporting-engine-12.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-12-0/reporting-engine-12-0-bi_sql_editor/

HviorForgeFlow committed:
[IMP] bi_sql_editor: black, isort, prettier

HviorForgeFlow committed:
[MIG] bi_sql_editor: Migration to 13.0

Saran440 committed:
[IMP] report_wkhtmltopdf_param: black, isort, prettier

Saran440 committed:
[13.0][MIG] report_wkhtmltopdf_param

OCA-git-bot committed:
bi_sql_editor 13.0.1.0.1

[UPD] README.rst

legalsylvain committed:
[WIP] imp bi_sql_editor

jprokop1 committed:
Translated using Weblate (Slovak)

Currently translated at 100.0% (21 of 21 strings)

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/sk/

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoice-reporting-12.0/account-invoice-reporting-12.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/account-invoice-reporting-12-0/account-invoice-reporting-12-0-base_comment_template/

feketemihai committed:
[MIG] [13.0] Migrate base_comment_template from account_invoic_reporting.

Commit:
[IMP] base_comment_template: Code refactorization

legalsylvain committed:
[FIX] bi_sql_editor : handle correctly translation for action name

pedrobaeza committed:
[MIG+FIX] base_comment_template: Fix migration scripts

As the field was not properly converted to property at 2.0.0, the script fails to do its
job when migrating from v11, so we need to adapt the scripts to this situation.

petrus-v committed:
[IMP] base_comment_template: black, isort, prettier

petrus-v committed:
[MIG] base_comment_template: Migration to 14.0 add unit test

[UPD] Update base_comment_template.pot

[UPD] README.rst

fshah-initos committed:
[14.0][MIG]report_wkhtmltopdf_param

CarlosRoca13 committed:
[IMP] base_comment_template: Improved sentences on fields

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 38.3% (43 of 112 strings)

Translation: reporting-engine-13.0/reporting-engine-13.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-13-0/reporting-engine-13-0-bi_sql_editor/es/

victoralmau committed:
[IMP] base_comment_template: Add function to get_comment_template records

victoralmau committed:
[FIX] base_comment_template: Fix tests: test_company_general_template

victoralmau committed:
[IMP] base_comment_template: Adapt to 14.0

[UPD] Update base_comment_template.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-14.0/reporting-engine-14.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-14-0/reporting-engine-14-0-base_comment_template/

victoralmau committed:
[FIX] base_comment_template: Change domain in rule according to company

victoralmau committed:
[IMP] base_comment_template: Refactor code and convert to Many2Many

pedrobaeza committed:
[FIX] base_comment_template: Incorrect m2m column names + script + avoid FK constraint

- The name of the columns were swapped, so better to fix it for avoiding
  mistakes due to this.
- Migration script for detecting the condition and swap column names.
- On v12 > v13 migration, include JOINs for avoiding FK constraint,
  which `ON CONFLICT` clause doesn't protect.

joao-p-marques committed:
[FIX] base_comment_template: Filter domain correctly

1. Safe eval domain before checking if it is defined. This makes [] be "False" on the first check, and provides the proper use case for when the domain is empty
2. Use filtered_domain in record instead of a search in self. When on a compute, we are on a NewId context, so the search would always fail

TT29309

victoralmau committed:
[FIX] base_comment_template: Delete res_partner.xml view (not used now)

base_comment_template 14.0.2.0.1

davejames committed:
[IMP] bi_sql_editor: black, isort, prettier

davejames committed:
bi_sql_editor: add support for group operators

this commit allows specifying the group operator for each field. This functionality was originally included in a seperate module "bi_sql_editor_aggregate" in version 12.

legalsylvain committed:
[IMP] bi_sql_editor : conserve cron settings, when setting materialized view to draft ; Set correct default values for cron and enable it

davejames committed:
bi_sql_editor: Migration to 14.0

oca-travis committed:
[UPD] Update bi_sql_editor.pot

OCA-git-bot committed:
[UPD] README.rst

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (36 of 36 strings)

Translation: reporting-engine-14.0/reporting-engine-14.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-14-0/reporting-engine-14-0-base_comment_template/pt/

mariadforgeflow committed:
[IMP] bi_sql_editor: black, isort, prettier

mariadforgeflow committed:
[MIG] bi_sql_editor: Migration to 15.0

legalsylvain committed:
[IMP] bi_sql_editor : add field context

feketemihai committed:
[14.0] [FIX] Fix base comment template rendering.

[UPD] Update base_comment_template.pot

[UPD] README.rst

base_comment_template 14.0.3.0.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-14.0/reporting-engine-14.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-14-0/reporting-engine-14-0-base_comment_template/

Commit:
[UPD] Update bi_sql_editor.pot

OCA-git-bot committed:
[UPD] README.rst

yankinmax committed:
[IMP] report_wkhtmltopdf_param: black, isort, prettier

yankinmax committed:
[IMP] report_wkhtmltopdf_param: pylint

yankinmax committed:
[MIG] report_wkhtmltopdf_param: Migration to 15.0

mariadforgeflow committed:
[FIX] bi_sql_editor: fix typo

OCA-git-bot committed:
bi_sql_editor 15.0.1.0.1

florian-dacosta committed:
[IMP] Make possible to make sql queries on an external database

lmignon committed:
[IMP] bi_sql_editor: Remove useless monkeypatch

On class creation by the ORM of 'manual' model, odoo now check if the table is a real table or a view to automatically set the '_auto' property on the class definition. We can therefore remove the monkey patch on 'ir.model'.

see https://github.com/odoo/odoo/commit/17c4f47b0ae7a0eec78b5555d52eeb0bb5e3cc5a

OCA-git-bot committed:
bi_sql_editor 15.0.1.0.2

Commit:
[FIX] sql_request_abstract: regression

pablontura committed:
Translated using Weblate (Catalan)

Currently translated at 41.0% (46 of 112 strings)

Translation: reporting-engine-15.0/reporting-engine-15.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-15-0/reporting-engine-15-0-bi_sql_editor/ca/

alan196 committed:
[MIG] base_comment_template: Migration to 15.0

gurneyalex committed:
IMP: bi_sql_editor lang

Allow to call refresh materialized view with an explicit lang passed in
the context and have this get priority over the user's lang

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (36 of 36 strings)

Translation: reporting-engine-14.0/reporting-engine-14.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-14-0/reporting-engine-14-0-base_comment_template/it/

etobella committed:
[FIX] base_comment_template: Typo error

base_comment_template 14.0.3.0.1

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 97.3% (109 of 112 strings)

Translation: reporting-engine-15.0/reporting-engine-15.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-15-0/reporting-engine-15-0-bi_sql_editor/ca/

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 94.1% (16 of 17 strings)

Translation: reporting-engine-15.0/reporting-engine-15.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-15-0/reporting-engine-15-0-report_wkhtmltopdf_param/ca/

victoralmau committed:
[MIG] base_comment_template: Migration to 15.0

TT35696

Commit:
[UPD] Update base_comment_template.pot

OCA-git-bot committed:
[UPD] README.rst

mhachraf committed:
[MIG] base_comment_template : Migration to 16.0

Commit:
[UPD] Update base_comment_template.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/

legalsylvain committed:
[MIG] bi_sql_editor : from 15.0 to 16.0
- move menu and action in according view file
- use new sql_request_abstract menu entry for SQL views configuration
- Use sql_request_abstract primary tree view
- Use sql_request_abstract primary form view
- move has_grouped_changed in sql_request_abstract module
- improve demo data
- add default count value to display in pivot, if no pivot_measures are defined ;
- set widget handle to sequence field
- reorder fields in tree view and add optional hide for some fields
- simplify : make field_description field required
- improve : add optional='hide' and 'show' on tree visibility
- replace obsolete base.menu_board_root by spreadsheet_dashboard entries
- update translation

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (57 of 57 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/es_AR/

victoralmau committed:
[FIX] base_comment_template: Add models text field to prevent error from ir.model model

TT40209

marylla committed:
Translated using Weblate (German)

Currently translated at 84.2% (48 of 57 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/de/

paulius-sladkevicius committed:
[IMP] bi_sql_editor manual field check performance

The `table_columns()` slows down modules loading significantly up to 2-3x.
In most cases `model._name.startswith(self._model_prefix)` returns `False`, so better to check it first.

Commit:
[UPD] Update base_comment_template.pot

OCA-git-bot committed:
base_comment_template 16.0.2.0.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (59 of 59 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (17 of 17 strings)

Translation: reporting-engine-15.0/reporting-engine-15.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-15-0/reporting-engine-15-0-report_wkhtmltopdf_param/es_AR/

Commit:
[UPD] Update bi_sql_editor.pot

OCA-git-bot committed:
[UPD] README.rst

yankinmax committed:
[MIG] report_wkhtmltopdf_param: Migration to 16.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-report_wkhtmltopdf_param/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 47.0% (8 of 17 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-report_wkhtmltopdf_param/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 14.2% (16 of 112 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 61.0% (36 of 59 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/it/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (112 of 112 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/es_AR/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 77.6% (87 of 112 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 16.9% (10 of 59 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 47.0% (8 of 17 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-report_wkhtmltopdf_param/sl/

Commit:
Update bi_sql_view.py

mymage committed:
Translated using Weblate (Italian)

Currently translated at 61.0% (36 of 59 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/it/

Commit:
[FIX] unlink bi_sql_view without cron

OCA-git-bot committed:
bi_sql_editor 16.0.1.0.3

mymage committed:
Translated using Weblate (Italian)

Currently translated at 14.2% (16 of 112 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/it/

OCA-git-bot committed:
bi_sql_editor 16.0.1.0.4

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 61.0% (36 of 59 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/it/

AaronHForgeFlow committed:
[FIX] base_comment_template: search when multiple models in a template

OCA-git-bot committed:
base_comment_template 16.0.2.0.1

alan196 committed:
[IMP] base_comment_template: allow to show code view in text field

alan196 committed:
[IMP] base_comment_template: update readme files

Change deprecated t-raw to t-out.
Rename invoice_comment_template to account_comment_template

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (59 of 59 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (112 of 112 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-report_wkhtmltopdf_param/es/

Commit:
[UPD] Update bi_sql_editor.pot

alan196 committed:
[IMP] base_comment_template: allow to select rendering engine in comment template

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update base_comment_template.pot

OCA-git-bot committed:
base_comment_template 16.0.2.1.0

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (63 of 63 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/es_AR/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (63 of 63 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (139 of 139 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/es/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/

Commit:
[UPD] Update bi_sql_editor.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-16.0/reporting-engine-16.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-bi_sql_editor/

HviorForgeFlow committed:
[IMP] pre-commit lint

leemannd committed:
[16.0][FIX][bi_sql_editor] unlink bi_sql_view without cron

Forward port of https://github.com/OCA/reporting-engine/pull/811

OCA-git-bot committed:
[BOT] post-merge updates

legalsylvain committed:
[IMP] bi_sql_editor : make installation idempotens

legalsylvain committed:
[REF] bi_sql_editor : simplify module tests

legalsylvain committed:
[REF] bi_sql_editor : Make test independant
[FIX] bi_sql_editor : allow to pass default values in copy() function

legalsylvain committed:
[FIX] bi_sql_editor : do not allow to create model (and sql views) if related model are not set on many2one fields. It prevents the error 'AttributeError: '_unknown' object has no attribute 'id''

legalsylvain committed:
[FIX] bi_sql_editor : use correct key word 'optional'. (and not 'option') in tree view

legalsylvain committed:
[IMP] bi_sql_editor : allow to reset to the previous state, and not only into 'draft' state

legalsylvain committed:
[REF] bi_sql_editor : simplification : Remove conditional display related to field_description, as this field is required and so, allways defined

legalsylvain committed:
[REM] bi_sql_editor : option_context_field field, as it is now possible to implement that feature with optional='hide' on the field list view

legalsylvain committed:
[IMP] bi_sql_editor : Allow to change fields settings if state is 'Model Valid'. Avoid to reset to draft and to have to recreate view that can take a while, if the view is materialized

legalsylvain committed:
[IMP] bi_sql_editor : Improve code coverage, adding various settings on field items

nguyenminhchien committed:
[IMP] bi_sql_editor: pre-commit auto fixes

Commit:
[IMP]base_comment_template: using markup.safe to render html in report

nguyenminhchien committed:
[MIG] sql_request_abstract: Migration to 17.0

nguyenminhchien committed:
[MIG] bi_sql_editor: Migration to 17.0

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 57.1% (36 of 63 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/it/

Commit:
[UPD] Update bi_sql_editor.pot

Commit:
[UPD] Update sql_request_abstract.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 88.2% (15 of 17 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-report_wkhtmltopdf_param/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (63 of 63 strings)

Translation: reporting-engine-16.0/reporting-engine-16.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-16-0/reporting-engine-16-0-base_comment_template/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 95.4% (42 of 44 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (140 of 140 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (44 of 44 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 19.2% (27 of 140 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (140 of 140 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/it/

trisdoan committed:
[IMP] report_wkhtmltopdf_param: pre-commit auto fixes

trisdoan committed:
[MIG] report_wkhtmltopdf_param: Migration to 17.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (140 of 140 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/it/

pedrobaeza committed:
[FIX] *: line too long

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

alan196 committed:
[IMP] base_comment_template: pre-commit auto fixes

alan196 committed:
[MIG] base_comment_template: Migration to 17.0

GuillemCForgeFlow committed:
[IMP]bi_sql_editor: add parent menu as an editable parameter when creating UI elements
When creating the UI for the SQL view, it is now possible to set a specific Parent Menu which will be used when creating the UI Menu for the report itself. The default value will be the `SQL Reports` menu.
Also, added the explanation on the `CONFIGURE.rst` file to have the description updated.

Commit:
[UPD] Update bi_sql_editor.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (144 of 144 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/it/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (144 of 144 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (16 of 16 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-report_wkhtmltopdf_param
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-report_wkhtmltopdf_param/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (44 of 44 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/sv/

Commit:
[UPD] Update bi_sql_editor.pot

Commit:
[UPD] Update sql_request_abstract.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-bi_sql_editor/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/

Commit:
[UPD] Update base_comment_template.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-base_comment_template/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (62 of 62 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-base_comment_template/it/

Commit:
[UPD] Update sql_request_abstract.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (62 of 62 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-base_comment_template/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (43 of 43 strings)

Translation: reporting-engine-17.0/reporting-engine-17.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-17-0/reporting-engine-17-0-sql_request_abstract/es_AR/

thienvh332 committed:
[MIG] bi_sql_editor: Migration to 18.0

thienvh332 committed:
[IMP] sql_request_abstract: apply pre-commit auto fixes

thienvh332 committed:
[MIG] sql_request_abstract: Migration to 18.0

thienvh332 committed:
add checklog-odoo.cfg

Commit:
[MIG] migrate report_wkhtmltopdf_param to 18.0

xaviedoanhduy commented:
> For being easily searchable and avoiding to duplicate efforts, please name the pull request following this pattern: 
`[18.0][MIG] <module>: Migration to 18.0.`

xaviedoanhduy commented on report_wkhtmltopdf_param/models/report_paperformat.py:
```diff
@@ -0,0 +1,43 @@
+            content = report._run_wkhtmltopdf(sample_html)
+            if not content:
+                raise ValidationError(
+                    _("Failed to create a PDF using the provided parameters.")
```
```suggestion
                    self.env._("Failed to create a PDF using the provided parameters.")
```

xaviedoanhduy commented on report_wkhtmltopdf_param/models/report_paperformat.py:
```diff
@@ -0,0 +1,43 @@
+    )
+
+    @api.constrains("custom_params")
+    def _check_recursion(self):
```
> Replace the deprecated _check_recursion() method by _has_cycle().

example how to used: 
https://github.com/odoo/odoo/blob/5a390fede312513a5c9b91d2d18d6d0cfdd43750/addons/account/models/account_tax.py#L542

LucasTran380381 replied:
change name to `_check_recursion_custom_params`

xaviedoanhduy replied:
sure, you can do it

xaviedoanhduy reviewed (changes requested):
I tested the functionality of the module and it works properly, but you should make some minor adjustments.

### result: 
![image](https://github.com/user-attachments/assets/9ac4bfb4-3b86-4901-99af-7bc24556f5b4)
![image](https://github.com/user-attachments/assets/d559a564-4801-45ca-a7cb-57ddbe5a1f8c)

xaviedoanhduy commented on report_wkhtmltopdf_param/tests/test_report_paperformat.py:
```diff
@@ -0,0 +1,38 @@
+        for report_paperformat in self.env["report.paperformat"].search([]):
+            with (
+                self.assertRaises(ValidationError),
+                self.assertLogs(level=logging.WARNING),
```
btw, if possible we should handle this `WARNING` instead of ignoring it

LucasTran380381 replied:
`WARNING v18c_report_wkhtmltopdf_param odoo.addons.base.models.ir_actions_report: wkhtmltopdf: Exit with code 1 due to network error: HostNotFoundError`

This warning due to `wkhtmltopdf` is trying to resolve "bad-parameter" as an host. Recommend to catch warning log in test-case (v17 has same warning log)

xaviedoanhduy reviewed (approved):
looks fine to me

xaviedoanhduy commented on report_wkhtmltopdf_param/models/report_paperformat.py:
```diff
@@ -36,8 +36,8 @@ def _check_recursion(self):
                 {"paperformat_id": paperformat.id}
             )
-            content = report._run_wkhtmltopdf(sample_html)
+            content = report._run_wkhtmltopdf([sample_html])
```
need a quote like:
> :param list[str] bodies: The html bodies of the report, one per page.

xaviedoanhduy reviewed (approved):
LGTM

OCA-git-bot committed:
Merge PR #937 into 18.0

Signed-off-by legalsylvain

Commit:
[UPD] Update sql_request_abstract.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-18.0/reporting-engine-18.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-18-0/reporting-engine-18-0-sql_request_abstract/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (45 of 45 strings)

Translation: reporting-engine-18.0/reporting-engine-18.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-18-0/reporting-engine-18-0-sql_request_abstract/it/

OCA-git-bot committed:
Merge PR #938 into 18.0

Signed-off-by HviorForgeFlow

Commit:
[UPD] Update bi_sql_editor.pot

Commit:
[UPD] Update sql_request_abstract.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-18.0/reporting-engine-18.0-sql_request_abstract
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-18-0/reporting-engine-18-0-sql_request_abstract/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-18.0/reporting-engine-18.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-18-0/reporting-engine-18-0-bi_sql_editor/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (146 of 146 strings)

Translation: reporting-engine-18.0/reporting-engine-18.0-bi_sql_editor
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-18-0/reporting-engine-18-0-bi_sql_editor/it/

yankinmax committed:
[IMP] base_comment_template: pre-commit auto fixes

yankinmax committed:
[MIG] base_comment_template: Migration to 18.0

OCA-git-bot committed:
Merge PR #961 into 18.0

Signed-off-by HviorForgeFlow

Commit:
[UPD] Update base_comment_template.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: reporting-engine-18.0/reporting-engine-18.0-base_comment_template
Translate-URL: https://translation.odoo-community.org/projects/reporting-engine-18-0/reporting-engine-18-0-base_comment_template/
